src/govuk/common.mjs

```diff
@@ -129,3 +129,53 @@ export function extractConfigByNamespace (configObject, namespace) {
   }
   return newObject
 }
+
+/**
+ * Normalise data
+ *
+ * 'If it looks like a duck, and it quacks like a duck…' 🦆
+ *
+ * If the passed value looks like a boolean or a number, convert it to a boolean
+ * or number.
+ *
+ * Designed to be used to convert config passed via data attributes (which are
+ * always strings) into something sensible.
+ *
+ * @param {String} value The value to normalise
+ * @returns {(string|boolean|number)} Normalised data
+ */
+export function normaliseData (value) {
+  switch (value) {
+    case 'true':
+      return true
+
+    case 'false':
+      return false
+
+    // Matches if converting to a number and back to a string results in the
+    // same value
+    case Number(value).toString():
```

Comment:
This does mean that these tests all pass as well…

```js
      expect(normaliseData('Infinity')).toEqual(Infinity)
      expect(normaliseData('-Infinity')).toEqual(-Infinity)
      expect(normaliseData('NaN')).toEqual(NaN)
```

I think that's probably fine, but flagging in case we think need to do anything about this.

Reply:
Ahh decimals are going to fail the **string-to-number-and-back™** equality check

```html
<element data-amount="12.90">
```

```mjs
console.log(Number('12.90').toString() === '12.90') // false
console.log(Number('12.90').toString() === '12.9') // true
```

Regarding `NaN`/`Infinity` you've got [`Number.isFinite()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isFinite) or the older [`isFinite()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/isFinite) global

>[Difference between Number.isFinite() and global isFinite()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isFinite#difference_between_number.isfinite_and_global_isfinite)
>In comparison to the global [isFinite()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/isFinite) function, this method doesn't first convert the parameter to a number. This means only values of the type number and are finite return true, and non-numbers always return false.

The nerd in me would want to use [`Number.isSafeInteger()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isSafeInteger) via [core-js polyfill](https://github.com/zloirock/core-js/blob/master/packages/core-js/modules/es.number.is-safe-integer.js) 😆

Reply:
That's a great point. Hard to tell how likely it is that users will want to pass decimals as config.

I do think that Infinity / NaN are more likely to occur due to an error in a service team's code, rather than being passed intentionally, and I don't think typing them as numbers is going to break things any more than if they were kept as strings?

I think we need to try and find the right balance between building out something that's robust enough for our future components, and ideally something that's reusable for bespoke components built by teams, but without slowing us down too much.

Personally, I'm inclined to say 'if it's good enough for Bootstrap, it's good enough for us…'. We could add a comment to note that it won't work for decimals and wait for this to come up as an issue?

Reply:
Yep, emphasis on _nerd_ 🙈 

Your call on this, will affect this one too:

```mjs
console.log(Number('0.5').toString() === '.5') // false
console.log(Number('0.5').toString() === '0.5') // true
```

Not a blocker, but a quick fix?

```mjs
console.log(isFinite('.5')) // true
console.log(isFinite('0.5')) // true
console.log(isFinite('12.90')) // true
console.log(isFinite('12.9')) // true

console.log(isFinite('NaN')) // false
console.log(isFinite('Infinity')) // false
console.log(isFinite('-Infinity')) // false
```

```suggestion
    case isFinite(value) && Math.abs(Number(value)) <= 0x1FFFFFFFFFFFFF:
```

Reply:
Updated with an alternative approach. Had to switch (hah!) to an if statement as we're no longer doing an equality check.

Also needed to cater for the empty string as `isFinite('') === true` and `isFinite('   ') === true`!

Reply:
Fab, just seen it. I'll reply on that line number!

(Well spotted, knew there'd be a catch using the more-supported `isFinite()` vs `Math.isFinite()`)